libs/blocks/form/form.js

```diff
@@ -0,0 +1,379 @@
+import sanitizeComment from '../../utils/sanitizeComment.js';
+
+const RULE_OPERATORS = {
+  equal: '=',
+  notEqual: '!=',
+  lessThan: '<',
+  lessThanOrEqual: '<=',
+  greaterThan: '>',
+  greaterThanOrEqual: '>=',
+};
+
+function createSelect(fd) {
+  const select = document.createElement('select');
+  select.id = fd.Field;
+  if (fd.Placeholder) {
+    const ph = document.createElement('option');
+    ph.textContent = fd.Placeholder;
+    ph.setAttribute('selected', '');
+    ph.setAttribute('disabled', '');
+    select.append(ph);
+  }
+  fd.Options.split(',').forEach((o) => {
+    const option = document.createElement('option');
+    const text = o.trim();
+    option.textContent = text;
+    option.value = text;
+    select.append(option);
+    if (fd.Default === text) select.value = text;
+  });
+  if (fd.Required === 'x') {
+    select.setAttribute('required', 'required');
+  }
+  return select;
+}
+
+function constructPayload(form) {
+  const payload = {};
+  [...form.elements].forEach((fe) => {
+    if (fe.type.match(/(?:checkbox|radio)/)) {
+      if (fe.checked) {
+        payload[fe.name] = payload[fe.name] ? `${fe.value}, ${payload[fe.name]}` : fe.value;
+      } else if (!payload[fe.name] && fe.closest('.group-container').classList.contains('required')) {
+        payload[fe.name] = 'required-not-checked';
+      }
+      return;
+    }
+    payload[fe.id] = fe.value;
+  });
+  return payload;
+}
+
+async function submitForm(form) {
+  const payload = constructPayload(form);
+  const keys = Object.keys(payload);
+  payload.timestamp = new Date().toJSON();
+  for (const key of keys) {
+    if (payload[key] === 'required-not-checked') {
+      const el = form.querySelector(`input[name="${key}"]`);
+      el.setCustomValidity('This box must be checked');
+      el.reportValidity();
+      const cb = () => {
+        el.setCustomValidity('');
+        el.reportValidity();
+        el.removeEventListener('input', cb);
+      };
+      el.addEventListener('input', cb);
+      return false;
+    }
+    payload[key] = sanitizeComment(payload[key]);
+  }
+  /* c8 ignore next 7 */
+  const resp = await fetch(form.dataset.action, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ data: payload }),
+  });
+  await resp.text();
+  return payload;
+}
+
+function clearForm(form) {
+  [...form.elements].forEach((fe) => {
+    if (fe.type.match(/(?:checkbox|radio)/)) {
+      fe.checked = false;
+    } else {
+      fe.value = '';
+    }
+  });
+}
+
+function createButton(fd, thankYou) {
+  const button = document.createElement('button');
```

Comment:
```suggestion
  const button = createTag('button', { class: 'button' }, fd.Label)
```
You could condense a number of lines thru out this file w/ the createTag func.